File: 389-Find_the_Difference/Find_the_Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int[] numOfChars = new int[26];
        
        for(int i = 0 ; i < s.length(); i++){
            numOfChars[s.charAt(i) - 'a']++;
        }
        
        for(int i = 0 ; i < t.length(); i++){
            if(numOfChars[t.charAt(i) - 'a'] == 0){
                return t.charAt(i);
            }
            
            numOfChars[t.charAt(i) - 'a']--;
        }
        
        return 'a';
    }
}

File: 0389-FindTheDifference/FindTheDifference.go
package findthedifference

func findTheDifference(s string, t string) byte {
	numOfChars := make([]int, 26)

	for i := range s {
		idx := s[i] - 'a'
		numOfChars[idx]++
	}

	for i := range t {
		idx := t[i] - 'a'
		if numOfChars[idx] == 0 {
			return t[i]
		}
		numOfChars[idx]--
	}

	return 'a'
}
